src/tpt_reports/report_generator.py

```diff
@@ -0,0 +1,414 @@
+"""Generate a TPT report using a passed data dictionary."""
+
+# Standard Python Libraries
+from datetime import datetime
+import os
+import secrets
+
+# Third-Party Libraries
+import numpy as np
+import pandas as pd
+from reportlab.lib import utils
+from reportlab.lib.colors import HexColor
+from reportlab.lib.pagesizes import letter
+from reportlab.lib.styles import ParagraphStyle
+from reportlab.lib.units import inch
+from reportlab.pdfbase import pdfmetrics
+from reportlab.pdfbase.ttfonts import TTFont
+from reportlab.platypus import (
+    Image,
+    KeepTogether,
+    PageBreak,
+    Paragraph,
+    Spacer,
+    Table,
+    TableStyle,
+)
+from reportlab.platypus.doctemplate import (
+    BaseDocTemplate,
+    NextPageTemplate,
+    PageTemplate,
+)
+from reportlab.platypus.frames import Frame
+
+BASE_DIR = os.path.abspath(os.path.dirname(__file__))
+TODAYS_DATE = datetime.today().strftime("%m/%d/%Y")
+REPORT_KEY_LENGTH = 17
+
+# Set fonts to be used in the PDF
+pdfmetrics.registerFont(
+    TTFont("Franklin_Gothic_Book", f"{BASE_DIR}/fonts/FranklinGothicBook.ttf")
+)
+
+pdfmetrics.registerFont(
+    TTFont(
+        "Franklin_Gothic_Medium_Regular",
+        f"{BASE_DIR}/fonts/FranklinGothicMediumRegular.ttf",
+    )
+)
+
+# Set page size
+defaultPageSize = letter
+PAGE_HEIGHT = defaultPageSize[1]
+PAGE_WIDTH = defaultPageSize[0]
+
+
+class MyDocTemplate(BaseDocTemplate):
+    """Extend the BaseDocTemplate to adjust Template."""
+
+    def __init__(self, filename, report_key, **kw):
+        """Initialize MyDocTemplate."""
+        # Set report_key as an environment variable
+        os.environ["TPT_REPORT_KEY"] = report_key
+        self.allowSplitting = 0
+        BaseDocTemplate.__init__(self, filename, **kw)
+        self.pagesize = defaultPageSize
+        self.encrypt = report_key
```

Comment:
Unless the order matters here for some reason, I'd reorder like this and alphabetize the attributes you're initializing:
```suggestion
        BaseDocTemplate.__init__(self, filename, **kw)
        self.allowSplitting = 0
        self.encrypt = report_key
        self.pagesize = defaultPageSize
```

Reply:
After testing additional json file inputs, we noticed that these attributes must be placed in this order in order to create the document correctly. [40a34bc](https://github.com/cisagov/tpt-reports/pull/15/commits/40a34bcc0b762c88e4fed1fc725deeb57e70cfd3)

Reply:
In that case, you should add a comment mentioning that the order matters here and why.

Reply:
Sure, no problem. The below illustrates why order matters.
self.allowSplitting must be called before BaseDocTemplate is initiated otherwise content written to the pdf that spills over will be too large for the frame or page. Some content is based on how large the input data is. The error message is"reportlab.platypus.doctemplate.LayoutError" and adds that it is "too large on page 4 in frame".
self.encrypt must be called after BaseDocTemplate otherwise the pdf opens without encryption. The pdf does not prompt for a password.
self.pagesize doesn't require special placement but comes after the others in the alphabet.

Reply:
Thanks @ameliav for the explanation. 👍  Once you've added a brief comment to the code summarizing all of that, I'll resolve this conversation.

Reply:
Comments added in commit: [603b20a](https://github.com/cisagov/tpt-reports/pull/15/commits/603b20a195f2ff03efa23a3bfb179c20dd4ccc48)